### PR TITLE
Exercise

### PR DESCRIPTION
**Pull request by pob31**
filled items #1 to #10

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -200,8 +200,12 @@ bool LeftHandDriving = true;
 float volumeGasTank = 70.0f;
```
Comment (matkatmusic) [inline main.cpp:200]: Why isn't this indented like the rest of your class members?
- [ ] fix the formatting.

File: main.cpp [hunk 2]
```diff
@@ -200,8 +200,12 @@ bool LeftHandDriving = true;
...
 
 //    1)  start
+    bool startEngine (); // returns the stage of the engine, whether it's running or not.
```
Comment (matkatmusic) [inline main.cpp:203]: no spaces between function name and argument list
- [ ] fix the formatting.

File: main.cpp [hunk 3]
```diff
@@ -220,7 +224,8 @@ float volumeGasTank = 70.0f;
 
 struct dogHead
```
Comment (matkatmusic) [inline main.cpp:225]: Types always begin with a capital letter
- [ ] fix the formatting.

File: main.cpp [hunk 4]
```diff
@@ -452,21 +477,49 @@ struct Compressor
...
+    struct EightFloats
+    {
+        float v1 = 0.0f;
+        float v2 = 0.0f;
+        float v3 = 0.0f;
+        float v4 = 0.0f;
+        float v5 = 0.0f;
+        float v6 = 0.0f;
+        float v7 = 0.0f;
+        float v8 = 0.0f;
```
Comment (matkatmusic) [inline main.cpp:489]: You can D.R.Y. this a bit:
``` 
float v1 = 0.0f,
      v2 = 0.0f,
      v3 = 0.0f,
```
etc...
you'll still need a semi-colon after the last one.
- [ ] fix the formatting.

File: main.cpp [hunk 5]
```diff
@@ -452,21 +477,49 @@ struct Compressor
...
+    struct EightInts
+    {
+        int v1 = 0.0f;
+        int v2 = 0.0f;
+        int v3 = 0.0f;
+        int v4 = 0.0f;
+        int v5 = 0.0f;
+        int v6 = 0.0f;
+        int v7 = 0.0f;
+        int v8 = 0.0f;
```
Comment (matkatmusic) [inline main.cpp:501]: D.R.Y.
- [ ] fix the formatting.

File: main.cpp [hunk 6]
```diff
@@ -485,21 +538,37 @@ struct Routing
...
+    struct EightFloats
+    {
+        float v1 = 0.0f;
+        float v2 = 0.0f;
+        float v3 = 0.0f;
+        float v4 = 0.0f;
+        float v5 = 0.0f;
+        float v6 = 0.0f;
+        float v7 = 0.0f;
+        float v8 = 0.0f;
```
Comment (matkatmusic) [inline main.cpp:550]: Give `OutputChannel` an instance of `Routing::EightFloats` as a member variable, or just make `EightFloats` a struct that isn't nested. 
- [ ] D.R.Y.

File: main.cpp (hunk 1 above)
Comment (pob31) [reply main.cpp]: Confusion with comments starting on beginning of the line. Fixed indentation.

File: main.cpp (hunk 2 above)
Comment (pob31) [reply main.cpp]: fixed

File: main.cpp (hunk 3 above)
Comment (pob31) [reply main.cpp]: fixed

File: main.cpp (hunk 4 above)
Comment (pob31) [reply main.cpp]: Streamlined as shown. Thanks for the tip! :)

File: main.cpp (hunk 5 above)
Comment (pob31) [reply main.cpp]: streamlined

File: main.cpp (hunk 6 above)
Comment (pob31) [reply main.cpp]: Moved both EightFloats and EightInts to code blocks of their own instead of being repeatedly nested.

**Pull request merged**